### PR TITLE
fix issue for seek to position

### PR DESCRIPTION
**Pull request by AmirhosseinShk**
on windows get nullPointer error when seek to position because get error when convert from int to int64 some times for small values and fix it now.

### CODE REVIEW

**Comment by bdlukaa** [comment]
it sometimes fail when running the example app:

```terminal
[ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: PlatformException(position, the position argument can not be null, null, null)
#0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
#1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
<asynchronous suspension>
#2      MethodChannelAudioPlayer.seek (package:just_audio_platform_interface/method_channel_just_audio.dart:152:34)
<asynchronous suspension>
#3      AudioPlayer.seek (package:just_audio/just_audio.dart:1070:9)
<asynchronous suspension>
```

I believe that, if `int` fails, we should lookup for `int_64` as fallback

**Comment by AmirhosseinShk** [comment]
huum, interesting , I'm agree with you , let's add int_64 as fall back and i think its could be ok

File: just_audio_windows/windows/player.hpp [hunk 1]
```diff
@@ -222,8 +222,11 @@ class AudioPlayer
         } else if (method_call.method_name().compare("setPreferredPeakBitRate") == 0) {
           result->Success(flutter::EncodableMap());
         } else if (method_call.method_name().compare("seek") == 0) {
-          const auto* position = std::get_if<int64_t>(ValueOrNull(*args, "position"));
-
+          auto* position = std::get_if<int>(ValueOrNull(*args, "position"));
+          
+          if (position == nullptr) {
+            position = std::get_if<int_64>(ValueOrNull(*args, "position"));
```
Comment (bdlukaa) [inline just_audio_windows/windows/player.hpp]: int_64 isn't a type.

```suggestion
            position = std::get_if<int64_t>(ValueOrNull(*args, "position"));
```

**Review by bdlukaa — CHANGES_REQUESTED** [review]

File: just_audio_windows/windows/player.hpp [hunk 2]
```diff
@@ -222,8 +222,11 @@ class AudioPlayer
         } else if (method_call.method_name().compare("setPreferredPeakBitRate") == 0) {
           result->Success(flutter::EncodableMap());
         } else if (method_call.method_name().compare("seek") == 0) {
-          const auto* position = std::get_if<int64_t>(ValueOrNull(*args, "position"));
-
+          auto* position = std::get_if<int>(ValueOrNull(*args, "position"));
+          
+          if (position == nullptr) {
+            position = std::get_if<int64_t>(ValueOrNull(*args, "position"));
```
Comment (bdlukaa) [inline just_audio_windows/windows/player.hpp]: the `position` parameter can not be changed type

**Review by bdlukaa — COMMENTED** [review]
I guess it'd be better if we handle them separatedely!

File: just_audio_windows/windows/player.hpp (hunk 2 above)
Comment (AmirhosseinShk) [reply just_audio_windows/windows/player.hpp]: yeah , that my mistake let me separate them and fix it

**Review by bdlukaa — APPROVED** [review]

File: just_audio_windows/windows/player.hpp (hunk 2 above)
Comment (VCasecnikovs) [reply just_audio_windows/windows/player.hpp]: @bdlukaa, hi, when this update will be available in the pub.dev package?

Comment (VCasecnikovs) [reply just_audio_windows/windows/player.hpp]: seek still throws an error

Comment (VCasecnikovs) [reply just_audio_windows/windows/player.hpp]: Unhandled Exception: PlatformException(position, the position argument can not be null, null, null)

Comment (AmirhosseinShk) [reply just_audio_windows/windows/player.hpp]: hi , idk when this update release on pub dev , but u can use master branch from git

can u tell me ur senario and input type? we test it alot on flutter windows with different type and it's work fine

**Pull request merged**